ipatests/test_integration/test_installation.py

```diff
@@ -1916,3 +1918,73 @@ def test_replica0_install(self):
         tasks.install_replica(
             self.master, self.replicas[0], setup_ca=False, setup_dns=False
         )
+
+
+class TestInstallwithSHA384withRSA(IntegrationTest):
+    num_replicas = 0
+
+    @classmethod
+    def install(cls, mh):
+        pass
+
+    def test_install_master_withalgo_sha384withrsa(self, server_cleanup):
+        tasks.install_master(
+            self.master,
+            extra_args=['--ca-signing-algorithm=SHA384withRSA'],
+        )
+
+        # check Signing Algorithm post installation
+        dashed_domain = self.master.domain.realm.replace(".", '-')
+        cmd_args = ['certutil', '-L', '-d',
+                    '/etc/dirsrv/slapd-{}/'.format(dashed_domain),
+                    '-n', 'Server-Cert']
+        result = self.master.run_command(cmd_args)
+        assert 'SHA-384 With RSA Encryption' in result.stdout_text
+
+    def test_install_master_modify_existing(self, server_cleanup):
```

Comment:
Is this something we are going to support? I guess changing the signing is just a capability of dogtag but do we need to test it? Do they support users doing this?

Reply:
Good question. I am not sure either. I thought about the possible scenarios and added tests for it.

Reply:
@ckelleyRH is tweaking the cert signing algorithm something that CS supports post-install?

Reply:
@rcritten I can't find any evidence that it is possible to do it. There's no REST endpoint that would allow you to do it, and I can't see a CLI way either. It looks like the signing algorithm is set from the config when the `CertDataInfo` object is initially created, and there are no other calls to that setter from anywhere.

Reply:
@ckelleyRH @rcritten After modifying CS.cfg and changing default Signature Algorithm and restarting services everything seems to work.